tests/test_lab_dev/test_states.py

```diff
@@ -0,0 +1,225 @@
+    """
+
+    @pytest.mark.parametrize("name", [None, "my_ket"])
+    @pytest.mark.parametrize("modes", [[0], [0, 1], [3, 19, 2]])
+    def test_init(self, name, modes):
+        state = Ket(name, modes)
+
+        assert state.name == (name if name else "")
+        assert state.modes == sorted(modes)
+        assert state.wires == Wires(modes_out_ket=modes)
+
+    def test_rshift(self):
+        ket = Coherent([0, 1], 1)
+        unitary = Dgate([0], 1)
+        u_component = CircuitComponent.from_attributes(
+            unitary.name, unitary.representation, unitary.wires
+        )
+        channel = Attenuator([1], 1)
+        ch_component = CircuitComponent.from_attributes(
+            channel.name, channel.representation, channel.wires
+        )
+
+        assert isinstance(ket >> unitary, Ket)
+        assert isinstance(ket >> channel, DM)
+        assert isinstance(ket >> unitary >> channel, DM)
+        assert isinstance(ket >> channel >> unitary, DM)
+        assert isinstance(ket >> u_component, CircuitComponent)
+        assert isinstance(ket >> ch_component, CircuitComponent)
+
+    def test_repr(self):
+        ket = Coherent([0, 1], 1)
+        ket_component = CircuitComponent.from_attributes(ket.name, ket.representation, ket.wires)
+
+        assert repr(ket) == "Ket(name=Coherent, modes=[0, 1])"
+        assert repr(ket_component) == "CircuitComponent(name=Coherent, modes=[0, 1])"
+
+
+class TestDM:
+    r"""
+    Tests for the ``DM`` class.
+    """
+
+    @pytest.mark.parametrize("name", [None, "my_dm"])
+    @pytest.mark.parametrize("modes", [[0], [0, 1], [3, 19, 2]])
+    def test_init(self, name, modes):
+        state = DM(name, modes)
+
+        assert state.name == (name if name else "")
+        assert state.modes == sorted(modes)
+        assert state.wires == Wires(modes_out_bra=modes, modes_out_ket=modes)
+
+    def test_rshift(self):
+        ket = Coherent([0, 1], 1)
+        unitary = Dgate([0], 1)
+        u_component = CircuitComponent.from_attributes(
+            unitary.name, unitary.representation, unitary.wires
+        )
+        channel = Attenuator([1], 1)
+        ch_component = CircuitComponent.from_attributes(
+            channel.name, channel.representation, channel.wires
+        )
+
+        dm = ket >> channel
+        assert isinstance(dm, DM)
+
+        assert isinstance(dm >> unitary >> channel, DM)
+        assert isinstance(dm >> channel >> unitary, DM)
+        assert isinstance(dm >> u_component, CircuitComponent)
+        assert isinstance(dm >> ch_component, CircuitComponent)
+
+    def test_repr(self):
+        ket = Coherent([0, 1], 1)
+        channel = Attenuator([1], 1)
+        dm = ket >> channel
+        dm_component = CircuitComponent.from_attributes(dm.name, dm.representation, dm.wires)
+
+        assert repr(dm) == "DM(name=None, modes=[0, 1])"
+        assert repr(dm_component) == "CircuitComponent(name=None, modes=[0, 1])"
+
+
+class TestCoherent:
```

Comment:
Just curious if we want to test the superposition of two coherent states, where to add that test?

Reply:
Integration? We can add an "Integration tests" folder on SisterPam now